tests/test.rs

```diff
@@ -454,6 +454,40 @@ fn feature_powerset() {
         .assert_stderr_not_contains("--features a,a");
 }
 
+#[test]
+fn powerset_deduplication() {
+    cargo_hack(["check", "--feature-powerset"])
+        .test_dir("tests/fixtures/powerset_deduplication")
+        .assert_success()
+        .assert_stderr_contains(
+            "
+            running `cargo check --no-default-features` on deduplication (1/15)
+            running `cargo check --no-default-features --features a` on deduplication (2/15)
+            running `cargo check --no-default-features --features b` on deduplication (3/15)
+            running `cargo check --no-default-features --features c` on deduplication (4/15)
+            running `cargo check --no-default-features --features a,c` on deduplication (5/15)
+            running `cargo check --no-default-features --features d` on deduplication (6/15)
+            running `cargo check --no-default-features --features a,d` on deduplication (7/15)
+            running `cargo check --no-default-features --features b,d` on deduplication (8/15)
+            running `cargo check --no-default-features --features c,d` on deduplication (9/15)
+            running `cargo check --no-default-features --features a,c,d` on deduplication (10/15)
+            running `cargo check --no-default-features --features e` on deduplication (11/15)
+            running `cargo check --no-default-features --features a,e` on deduplication (12/15)
+            running `cargo check --no-default-features --features c,e` on deduplication (13/15)
+            running `cargo check --no-default-features --features a,c,e` on deduplication (14/15)
```

Comment:
This should be excluded because `c` enables `a` via `b`, but I haven't yet implemented the detection of indirectly dependent features.

Reply:
Are features able to depend on each other? If not, some sort of recursion to get the full list of features enabled could work. Even if mutual dependencies are a thing, all that would be needed extra is a check to see if it's already been visited.

Reply:
> Are features able to depend on each other?

As far as I know, yes.

Reply:
> all that would be needed extra is a check to see if it's already been visited.

Fixed by using this way, thanks for the advice!